src/dioptra/restapi/v1/artifacts/schema.py

```diff
@@ -81,10 +128,40 @@ class ArtifactPageSchema(BasePageSchema):
     )
 
 
+class ShowFileTreeQueryParametersSchema(Schema):
+    """A schema for adding show_file_tree query parameters to an artifact
+    resource endpoint."""
+
+    showFileTree = fields.Boolean(
+        attribute="show_file_tree",
+        metadata=dict(description="Display files as a directory."),
+        load_default=None,
+    )
+
+
+class DownloadPathQueryParametersSchema(Schema):
+    """A schema for adding path and downlaod query parameters to a resource endpoint."""
+
+    path = fields.String(
+        attribute="path",
+        metadata=dict(description="Path of a specific artifact."),
+        load_default=None,
+    )
+    download = fields.Boolean(
+        attribute="download",
+        metadata=dict(
+            description="Determines whether the file will be downloaded or viewed."
+        ),
+        load_default=None,
+    )
+
+
 class ArtifactGetQueryParameters(
     PagingQueryParametersSchema,
     GroupIdQueryParametersSchema,
     SearchQueryParametersSchema,
```

Comment:
I think we should leave out Search and SortBy query parameters at least for now. These would require a custom implementation since search and sort for resources are implemented via the sqlalchemy.